### PR TITLE
refactor: update repository structure and configuration

### PR DESCRIPTION
**Pull request by RealShadowNova**
This PR represents a comprehensive restructuring and reconfiguration of the repository, aimed at preparing it for release. It sets the groundwork for ensuring a smooth and efficient development process. Further PRs will focus on reviewing source code to ensure correctness and thorough testing coverage.

## Changes

- **Dependency Management**: Removed [`husky`](https://github.com/typicode/husky) in favor of a Yarn plugin, reducing the maintenance burden and streamlining the codebase.
- **Workflow Enhancements**: Updated all workflows to ensure consistency across the repository.
- **Contributing Guide**: Removed the contributing guide, as GitHub now automatically directs contributors to our organization's standard guidelines at https://github.com/josh-development/.github.
- **Ownership**: Updated codeowners to reflect current contributors.
- **Package Updates**: Updated, fixed, and restructured various files in the `packages/` folder, including `.cliff-jumperrc.yml`, `cliff.toml`, `package.json`, `README.md`, and `tsconfig.eslint.json`.
- **Build Process**: Removed `rimraf` in favor of custom scripts for cleaning output files, optimizing our build process.
- **Commit Standards**: Removed `.commitlintrc.json` in favor of utilizing `@commitlint/config-conventional`, promoting adherence to conventional commit standards.
- **Configuration Updates**: Updated, fixed, and restructured almost all configuration files in the root folder for improved clarity and efficiency.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -7,28 +7,72 @@ on:
...
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4
+
+      - name: Setup Node & Install Dependencies
+        uses: josh-development/.github/setup-node@main
+        with:
+          turbo: true
+
       - name: Generate Documentation
-        run: yarn docs --cache-dir=./.turbo
+        run: yarn docs
```
Comment (dan-online) [inline .github/workflows/continuous-integration.yml:59]: Afaik the josh-development/setup-node depends on a .turbo directory for the cache

File: .yarn/plugins/@yarnpkg/plugin-git-hooks.cjs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/* eslint-disable */
```
Comment (dan-online) [inline .yarn/plugins/@yarnpkg/plugin-git-hooks.cjs:1]: Just checking that these are supposed to be pushed?

Comment (RealShadowNova) [reply .yarn/plugins/@yarnpkg/plugin-git-hooks.cjs:1]: Yes.

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (RealShadowNova) [reply .github/workflows/continuous-integration.yml:59]: Possibly resolved by setting the [`TURBO_CACHE_DIR`](https://turbo.build/repo/docs/reference/command-line-reference/run#--cache-dir) env in the https://josh-development/.github/blob/main/setup-node/action.yml file.

**Review by dan-online — APPROVED** [review]

**Pull request merged**